src/imio/restapi/serializer/base.py

```diff
@@ -52,17 +55,29 @@ def _additional_fields(self):
         """By default add 'id' and 'UID' to returned data."""
         return ["id", "UID"]
 
-    def _set_metadata_fields(self):
-        """Must be set in request.form."""
-        form = self.request.form
-        # manage metadata_fields
-        additional_metadata_fields = listify(form.get("metadata_fields", []))
+    def _get_metadata_fields_name(self):
+        """May be overrided when necessary."""
+        return "metadata_fields"
+
+    def metadata_fields(self):
+        """Override from plone.restapi to be able to change the metadata_fields name..."""
+        # XXX following line is replaced
+        # additional_metadata_fields = self.request.form.get("metadata_fields", [])
+        additional_metadata_fields = self.request.form.get(self._get_metadata_fields_name(), [])
+        if not isinstance(additional_metadata_fields, list):
+            additional_metadata_fields = [additional_metadata_fields]
+        # XXX begin custom additional metadata_fields
         additional_metadata_fields += self._additional_fields
-        # manage duplicates
-        form["metadata_fields"] = list(set(additional_metadata_fields))
+        # XXX end custom additional metadata_fields
```

Comment:
XXX should be removed since that indicate that something need to be changed

Reply:
salut @mpeeters 

j'ai simplifié les commentaires en enlevant les XXX

Merci pour le review!

Gauthier